src/service/nicknameService.ts

```diff
@@ -60,19 +62,20 @@ export function buildUserRank(member: GuildMember): UserRank {
 async function checkForChange(
     userRank: UserRank,
     user: UserData,
-    member: GuildMember
+    member: GuildMember,
+    isRoundOver: boolean
 ) {
     if (
         user.name != userRank.user_name ||
         user.rank != userRank.rank ||
-        buildRankNickname(user) != member.nickname
+        buildRankNickname(user, isRoundOver) != member.nickname
     ) {
-        await processMember(member, user, userRank.id);
+        await processMember(member, user, userRank.id, isRoundOver);
     }
 }
 
-function buildRankNickname(user: UserData) {
-    const rank = rankMap.get(user.rank) ?? "";
+function buildRankNickname(user: UserData, isRoundOver: boolean) {
+    const rank = isRoundOver ? "" : rankMap.get(user.rank) ?? "";
     return `${rank} ${user.name}`;
```

Comment:
this will result in a leading whitespace when `isRoundOver` is true

Reply:
no discord trims it, this is already what happens when user is not in round